src/Microsoft.AspNetCore.Mvc.Razor/Internal/ViewLocationCacheKey.cs

```diff
@@ -131,6 +141,7 @@ public override int GetHashCode()
             hashCodeCombiner.Add(ViewName, StringComparer.Ordinal);
             hashCodeCombiner.Add(ControllerName, StringComparer.Ordinal);
             hashCodeCombiner.Add(AreaName, StringComparer.Ordinal);
+            hashCodeCombiner.Add(PageName, StringComparer.Ordinal);
```

Comment:
Discussed, we added a test for the view engine that the caching works equally well for pages.